clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -109,29 +109,71 @@
+            MetadataRequest.schemaVersions(), MetadataResponse.schemaVersions()),
+    LEADER_AND_ISR(4, "LeaderAndIsr", true,
+            "Sent by the controller to other brokers to change the whether they are leaders or followers for " +
+            "the partitions given in the request.",
+            LeaderAndIsrRequest.schemaVersions(), LeaderAndIsrResponse.schemaVersions()),
+    STOP_REPLICA(5, "StopReplica", true,
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            StopReplicaRequest.schemaVersions(), StopReplicaResponse.schemaVersions()),
+    UPDATE_METADATA(6, "UpdateMetadata", true,
+            "Sent by the controller to other brokers in the cluster to get them to update their metadata about " +
+            "the cluster.",
+            UpdateMetadataRequest.schemaVersions(), UpdateMetadataResponse.schemaVersions()),
+    CONTROLLED_SHUTDOWN(7, "ControlledShutdown", true,
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            ControlledShutdownRequest.schemaVersions(), ControlledShutdownResponse.schemaVersions()),
+    OFFSET_COMMIT(8, "OffsetCommit",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            OffsetCommitRequest.schemaVersions(), OffsetCommitResponse.schemaVersions()),
+    OFFSET_FETCH(9, "OffsetFetch",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            OffsetFetchRequest.schemaVersions(), OffsetFetchResponse.schemaVersions()),
+    FIND_COORDINATOR(10, "FindCoordinator",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            FindCoordinatorRequest.schemaVersions(), FindCoordinatorResponse.schemaVersions()),
+    JOIN_GROUP(11, "JoinGroup",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            JoinGroupRequest.schemaVersions(), JoinGroupResponse.schemaVersions()),
+    HEARTBEAT(12, "Heartbeat",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            HeartbeatRequest.schemaVersions(), HeartbeatResponse.schemaVersions()),
+    LEAVE_GROUP(13, "LeaveGroup",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            LeaveGroupRequest.schemaVersions(), LeaveGroupResponse.schemaVersions()),
+    SYNC_GROUP(14, "SyncGroup",
+            null,//TODO write a brief description of the participants and purpose of this message exchange
+            SyncGroupRequest.schemaVersions(), SyncGroupResponse.schemaVersions()),
+    DESCRIBE_GROUPS(15, "DescribeGroups",
+            "Sent by an admin client to a broker to get detailed information (such as group membersip) about a specific consumer group.",
```

Comment:
Just a minor nit. Using "(admin)" instead of "admin" as you have already done for admin client related messages.